tests/worker/test_jobs.py

```diff
@@ -428,65 +520,421 @@ async def test_create_mapped_variants_for_scoreset_with_existing_mapped_variants
+        result = await map_variants_for_score_set(standalone_worker_context, uuid4().hex, score_set.id, 1)
+
+    # TODO: How are errors persisted? Test persistence mechanism.
+    score_set = session.scalars(select(ScoreSetDbModel).where(ScoreSetDbModel.urn == score_set.urn)).one()
+    mapped_variants_for_score_set = session.scalars(
+        select(MappedVariant).join(Variant).join(ScoreSetDbModel).filter(ScoreSetDbModel.urn == score_set.urn)
+    ).all()
+
+    assert (await standalone_worker_context["redis"].llen(MAPPING_QUEUE_NAME)) == 0
+    assert (await standalone_worker_context["redis"].get(MAPPING_CURRENT_ID_NAME)).decode("utf-8") == ""
+    assert not result["success"]
+    assert not result["retried"]
+    assert len(mapped_variants_for_score_set) == 0
+    assert score_set.mapping_state == MappingState.failed
+    assert score_set.mapping_errors is not None
 
 
-@pytest.mark.skip
 @pytest.mark.asyncio
 async def test_create_mapped_variants_for_scoreset_mapping_exception(
     setup_worker_db, async_client, standalone_worker_context, session, data_files
 ):
-    async def awaitable_http_error():
-        raise HTTPError
+    score_set = await setup_records_files_and_variants(
+        session,
+        async_client,
+        data_files,
+        TEST_MINIMAL_SEQ_SCORESET,
+        standalone_worker_context,
+    )
+    # We are skipping the manager in these cases to test directly on the mapping job,
+    # so pop from the queue as if the manager had executed.
+    await sanitize_mapping_queue(standalone_worker_context, score_set)
+
+    # We seem unable to mock requests via requests_mock that occur inside another event loop. Workaround
+    # this limitation by instead patching the _UnixSelectorEventLoop 's executor function, with a coroutine
+    # object that sets up test mappingn output.
+    #
+    # TODO: Does this work on non-unix based machines?
+    # TODO: Is it even a safe operation to patch this event loop method?
+    with patch.object(
+        _UnixSelectorEventLoop,
+        "run_in_executor",
+        return_value=awaitable_exception(),
+    ):
+        result = await map_variants_for_score_set(standalone_worker_context, uuid4().hex, score_set.id, 1)
 
+    # TODO: How are errors persisted? Test persistence mechanism.
+    score_set = session.scalars(select(ScoreSetDbModel).where(ScoreSetDbModel.urn == score_set.urn)).one()
+    mapped_variants_for_score_set = session.scalars(
+        select(MappedVariant).join(Variant).join(ScoreSetDbModel).filter(ScoreSetDbModel.urn == score_set.urn)
+    ).all()
+
+    assert (await standalone_worker_context["redis"].llen(MAPPING_QUEUE_NAME)) == 1
+    assert (await standalone_worker_context["redis"].get(MAPPING_CURRENT_ID_NAME)).decode("utf-8") == ""
+    assert not result["success"]
+    assert result["retried"]
+    assert len(mapped_variants_for_score_set) == 0
+    assert score_set.mapping_state == MappingState.queued
+    assert score_set.mapping_errors is not None
+
```

Comment:
Would it be desirable to test that the retry succeeds?

Reply:
I guess my thought is that we are testing that we are in a state where the retry is ready to be executed, and that the manager is the thing which is actually executing the retry. It's a little weird here because we never actually use ARQ as a queue in the tests, we just test the jobs directly, so the interaction between jobs isn't well tested (which is what you are rightly getting at).

In that vein, I'm going to add five more tests that are more end to end than the existing ones and I think cover the possible job flows:
- Does the variant creation process run end to end successfully (create variants, enqueue a manager process, and map variants)?
- Does the variant creation process halt successfully (create variants unsuccesfully, do not enqueue a manager process)?
- Does the manager execute, queue a job, and allow it to finish successfully?
- Does the manager execute, queue a job (which we will force to fail), retry and re-queue the job, and then complete successfully on the next attempt?
- Does the manager execute, queue a job which fails multiple times as desired, and then exit its execution successfully?

These tests will test that the worker after each job execution is correct, while the new tests will test that the workflow is executing properly.